### ISSUE TITLE
geojson pick confusion when set perPositionHeight

### ISSUE THREAD

**Issue by znxd-wh** [CesiumGS/cesium#10091]
hello

I use GeoJsonDataSource to load geojson data, when i set the perPositionHeight=false,single click on the map to pickup a geojson feathure, it may return a wrong result.

![image](https://user-images.githubusercontent.com/42760174/153555299-d4687bd4-8146-42a3-9ce4-9d980daac967.png)

when i set entity.polygon.perPositionHeight = false, it will happen,but if set entity.polygon.perPositionHeight = true, it works fine. this happens with cesium version 1.88, when change cesium version to 1.74,it works fine as well.

following is the code:
```js
var promise = Cesium.GeoJsonDataSource.load('data/wenmiao.json', { clampToGround: true });
      promise.then(function (dataSource) {
        // viewer.dataSources.add(dataSource);
        var entities = dataSource.entities.values;
        for (var i = 0; i < entities.length; i++) {
          var entity = entities[i];
          entity.polygon.perPositionHeight = false;
          entity.polygon.classificationType = Cesium.ClassificationType.BOTH;
          entity.heightReference = Cesium.HeightReference.CLAMP_TO_GROUND
          entity.polygon.material= Cesium.Color.BLUE;
          viewer.entities.add(entity)
        }
      });
      viewer.flyTo(promise);
      var scene = viewer.scene;

      var handler = new Cesium.ScreenSpaceEventHandler(viewer.scene.canvas);
      handler.setInputAction(function (movement) {
        var pick = viewer.scene.pick(movement.position);
        if(pick){
          selectedEntity = pick.id;
          pick.id.polygon.material = new Cesium.Color(255, 0, 0, 1)
        }
      }, Cesium.ScreenSpaceEventType.LEFT_CLICK);
```

**Comment by ggetz** [comment]
@ebogo1 Could you triage this issue by  reproducing and seeing if you can find the probable cause?

**Comment by ebogo1** [comment]
@znxd-wh I wasn't able to reproduce this with the GeoJson examples we have in our Sandcastle gallery. Are you able to share the data you were testing this with? Thanks!

**Comment by znxd-wh** [comment]
@ebogo1  attach is my code with data, unzip it and publish with a web container such tomcat,iis,nginx. visit the index.html page,you will see the problem. click a feather several times, different feature will be picked.
![image](https://user-images.githubusercontent.com/42760174/153978876-5c0fb45f-5da8-4dac-925b-50d9df1742ef.png)

cesium 1.88 does not work,but 1.74 works fine. but if i don't set entity.polygon.perPositionHeight = false,both version work well,when i set entity.polygon.perPositionHeight = false,only version 1.74 works. 

look forward to your response and thanks 

[geojson.zip](https://github.com/CesiumGS/cesium/files/8065155/geojson.zip)

**Comment by ebogo1** [comment]
This looks like a regression in 1.80 from https://github.com/CesiumGS/cesium/pull/9398. The bug started happening on the first [commit](https://github.com/CesiumGS/cesium/pull/9398/commits/071a6a8bb2325ba4b014641ba6e5333f2891a00e) - it wasn't immediately obvious where the bug is in those changes but it looks like something with clamping broke the `perPositionHeight` behavior when set to false. 

- [Local Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=dVRhT9swEP0rVr6QisqBSXxpCxoU1nYrFNGyfVimyjiX9oZjR7YT1E3899lJaEPppEqV79579+58Tsk0KRFeQJNzIuGFDMFgkdHvVSw84tVxqKRlKEEfdcnfWBKCGVuB3txrVWICutemXmo+QnPL8jnoEvTkPTSs+IQUWvRIHKytzU0viozDoitGmeYrNEoKV45ylUVebjKPNBi7RUU/lBbJspGOtrXiwIu/dmL52unHMpalay/XKkMDrr/G4AjUV1fhmlk2V4XmQIViSXhEo8SFoheQGTJFfzuM75dwwbJ8oUZaFTLpEasLIJV+o0ztGmSYFpJbVJKEyVa4U0/LuwBp0SIYZ2OXp29RWjJRgOl7dKo0CT0FHfak7/4GWzYVIFd27YLHx414S37jCG/In/irX6frFM2V2KyUpDnoe2XQWx0DrtbWkVImDByGu+aNwRQ584zFJm8NcvghR69mi/F7oXVV5AFS0CB5iz1+n6DD6eXt/XIxW44eZo9314ftZMyCRiZaJpRQml5NH28aRr3Ou9GyJAlrlU6FeN1tR41MxWahwuYyfcYP1HCQ3myDqY7blVozmYj9JzPnGkDOc8bhpnQFxzUobCtQzpyA8UUaDWrATmRe2Mtqe1prlKkSMqfTWqIc+fOeJ+pjW6ybU321dauYktDnt6viDxSTQ+NsdVKNNPx0dtYlJ9XvdDe57v/ara5/evNlsRxOJ8NvnhF0g4GxGwEXdXVCPmOWK2396w8pjSxkuXAWTPRU8GewlBvTaa5xELWpgwRLgsl5HOx9keKAVCvqMmkhxBz/QBxcDCKH/0D1zxzlaua+FIJtPGx9ejGtg5TSQeSOh5lWKfHE9J7yPw) 
- [GeoJson for testing](https://github.com/CesiumGS/cesium/files/8074529/wenmiao.json.txt)

**Comment by znxd-wh** [comment]
hi @ebogo1 

did you mean this bug has appeared since version 1.80？so do you have any idea to fix it? or i do not set the perPositionHeight parameter, but can you please tell me a method to let geojson polygon clamp on 3dtiles model?like the below effect.
![image](https://user-images.githubusercontent.com/42760174/154447601-c53e4469-cea4-461c-930c-975837b72f22.png)

**Comment by ebogo1** [comment]
@znxd-wh Yeah, this bug was introduced in 1.80. I edited my first comment with more details. The shader affects all classification types so it's tricky to work out a quick fix. If we have a chance to revisit the issue we'll post updates here.

**Comment by ShenWeiQun** [comment]
I've also had this issue recently
For the time being, you can consider the following ways to solve the problem
```diff
handler.setInputAction(function (movement) {
-  var pick = viewer.scene.pick(movement.position);
-  if (pick) {
-    selectedEntity = pick.id;
-    pick.id.polygon.material = new Cesium.Color(255, 0, 0, 1)
-  }
+  const pickedObjects = scene.drillPick(movement.position);
+  if (pickedObjects) {
+    var pick = pickedObjects.pop();
+    selectedEntity = pick.id;
+    pick.id.polygon.material = new Cesium.Color(255, 0, 0, 1)
+  }
}, Cesium.ScreenSpaceEventType.LEFT_CLICK);
```
